vmdb/app/controllers/container_controller.rb

```diff
@@ -0,0 +1,360 @@
+    end
+
+    if @edit && @edit.fetch_path(:adv_search_applied, :qs_exp) # If qs is active, save it in history
+      x_history_add_item(:id     => x_node,
+                         :qs_exp => @edit[:adv_search_applied][:qs_exp],
+                         :text   => @right_cell_text)
+    else
+      x_history_add_item(:id => treenodeid, :text => @right_cell_text)  # Add to history pulldown array
+    end
+
+    # After adding to history, add name filter suffix if showing a list
+    unless @search_text.blank?
+      @right_cell_text += _(" (Names with \"%s\")") % @search_text
+    end
+  end
+
+  #set partial name and cell header for edit screens
+  def set_right_cell_vars(action)
+    name = @record ? @record.name.to_s.gsub(/'/,"\\\\'") : "" # If record, get escaped name
+    case action
+    when "container_edit"
+      partial = "container_form"
+      header = _("Editing %{model} \"%{name}\"") % {:name => @record.name,
+                                                    :model => ui_lookup(:model => "Container")}
+      action = "container_edit"
+    else
+      action = nil
+    end
+    return partial,action,header
+  end
+
+  # Replace the right cell of the explorer
+  def replace_right_cell(action = nil, replace_trees = [])
+    @explorer = true
+    partial, action_url, @right_cell_text = set_right_cell_vars(action) if action # Set partial name, action and cell header
+    get_node_info(x_node) if !@in_a_form && !params[:display]
+    replace_trees = @replace_trees if @replace_trees  # get_node_info might set this
+    type, _ = x_node.split("_").last.split("-")
+    trees = {}
+    if replace_trees
+      trees[:containers] = build_containers_tree if replace_trees.include?(:containers)
+    end
+    record_showing = type && ["Container"].include?(TreeBuilder.get_model_for_prefix(type))
+    if !@in_a_form && !@sb[:action]
+      v_buttons, v_xml = build_toolbar_buttons_and_xml("x_gtl_view_tb") unless record_showing
+      c_buttons, c_xml = build_toolbar_buttons_and_xml(center_toolbar_filename)
+    end
+    h_buttons, h_xml = build_toolbar_buttons_and_xml("x_history_tb") unless @in_a_form
+
+    render :update do |page|
+      # Build hash of trees to replace and optional new node to be selected
+      trees.each do |tree_name, tree|
+        page.replace(
+          "#{tree_name}_tree_div",
+          :partial => 'shared/tree',
+          :locals  => {
+            :tree => tree,
+            :name => tree.name
+          }
+        )
+      end
+      page << "dhxLayoutB.cells('b').setText('#{escape_javascript(h(@right_cell_text))}');"
+
+      # Replace right cell divs
+      if ["container_edit"].include?(action)
+        page.replace_html("main_div", :partial=>partial)
+      elsif params[:display]
+        partial_locals = Hash.new
+        partial_locals[:controller] = "container"
+        partial = "layouts/x_gtl"
+        if partial == "layouts/x_gtl"
+          partial_locals[:action_url] = @lastaction
```

Comment:
Since this code will always execute, you can change line 247 to be:

```ruby
partial_locals = { :controller => "container", :action_url => @lastaction }
```

and remove this line as well as 248.